src/connector/codec/src/decoder/protobuf/parser.rs

```diff
@@ -61,20 +61,24 @@ fn pb_field_to_col_desc(
     if let Kind::Message(m) = field_descriptor.kind()
         && !messages_as_jsonb.contains(m.full_name())
     {
-        let field_descs = if let DataType::List { .. } = field_type {
+        // TODO(struct): assign type name once supported.
+        let _type_name = m.full_name().to_owned();
+        // TODO(struct): since we deprecated `field_descs` in `ColumnDesc`, there's no need to
+        // traverse the fields here except for maintaining the same column id (index) as the
+        // original implementation.
+        let _field_descs = if let DataType::List { .. } = field_type {
```

Comment:
As a result, there's no information for the "column" id of the nested fields, which seems to make resolving https://github.com/risingwavelabs/risingwave/issues/19755 even more challenging. I may reintroduce it in future PRs, but in a clearer manner.

Reply:
Just to confirm, does it mean that we will only be able to alter subfields for newly created structs? 🤔

Reply:
Not that true as if we really want this to be available on existing tables, we can introduce a "mini" multi-versioned schema for the `STRUCT` type when we are going to upgrade its storage representation.